Inc/HALALMock/Services/Logger/Logger.hpp

```diff
@@ -0,0 +1,146 @@
+    do {                                                        \
+        Logger::set_metadata(__LINE__, __FUNCTION__, __FILE__); \
+        if (hasFlag(Log::config, Logger::LogLevel::DEBUG))      \
+            Logger::log(x, "DEBUG", ANSI_COLOR_GREY);           \
+    } while (0)
+
+#define LOG_INFO(x)                                             \
+    do {                                                        \
+        Logger::set_metadata(__LINE__, __FUNCTION__, __FILE__); \
+        if (hasFlag(Log::config, Logger::LogLevel::INFO))       \
+            Logger::log(x, "INFO", ANSI_COLOR_GREEN);           \
+    } while (0)
+
+#define LOG_WARNING(x)                                          \
+    do {                                                        \
+        Logger::set_metadata(__LINE__, __FUNCTION__, __FILE__); \
+        if (hasFlag(Log::config, Logger::LogLevel::WARNING))    \
+            Logger::log(x, "WARNING", ANSI_COLOR_YELLOW);       \
+    } while (0)
+
+#define LOG_ERROR(x)                                            \
+    do {                                                        \
+        Logger::set_metadata(__LINE__, __FUNCTION__, __FILE__); \
+        if (hasFlag(Log::config, Logger::LogLevel::ERROR))      \
+            Logger::log(x, "ERROR", ANSI_COLOR_RED);            \
+    } while (0)
+
+#define LOG_FATAL(x)                                            \
+    do {                                                        \
+        Logger::set_metadata(__LINE__, __FUNCTION__, __FILE__); \
+        if (hasFlag(Log::config, Logger::LogLevel::FATAL))      \
+            Logger::log(x, "FATAL", ANSI_COLOR_MAGENTA);        \
+    } while (0)
+
+class FileManager {
+   public:
+    FileManager(const std::ios_base::openmode file_mode);
+    void add(const std::string& msg);
+    ~FileManager();
+
+   private:
+    std::ofstream file_stream;
+};
+
+/** @brief Class used to Log info in simulator mode.
+ * You can log in five levels: DEBUG, INFO, WARNING, ERROR and FATAL.
+ *
+ * To configure it, you must set in the extern variable config
+ * a mask 'oring' the bits that represents what you want.
+ * You can configure which level of logs you want (DEBUG, INFO...),
+ * if you want to write logs on the console and/or on a file,
+ * and if you want to erase or not the log file.
+ *
+ * To use it, you must use one of the five MACROS defined in this file,
+ * each one representing the level of the log you want to print.
+ *
+ * As a tip, you can use std::format to log formatted messages
+ */
+class Logger {
+   public:
+    enum class LogLevel : unsigned int {
+        DEBUG = 0b00001,
+        INFO = 0b00010,
+        WARNING = 0b00100,
+        ERROR = 0b01000,
+        FATAL = 0b10000,
+    };
+
+    static void log(const std::string& msg, const std::string& level,
+                    const char* colour);
+    static void set_metadata(int line, const char* function, const char* file);
```

Comment:
altough you are using the macros to log, this is still public and can be used standalone without the macro, this means someone could forget to call set_metadata before each call to log. Instead, I would add a parameter to log (called something like metadata) with a struct as its type, this struct will hold the metadata, like line, function, file, etc.

Reply:
this is just a comment, if you don't want to do this, it's fine, just resolve this conversation